src/javascript/SelectorTypes/SystemName/SystemName.jsx

```diff
@@ -11,14 +11,18 @@ import classes from './SystemName.scss';
 export const SystemNameCmp = ({field, value, values, id, editorContext, onChange}) => {
     const {t} = useTranslation();
 
+    const handleOnChange = value => {
+        onChange(replaceSpecialCharacters(value)?.toLowerCase());
+    };
+
     return (
         <>
             <Text
                 field={{...field, readOnly: field.readOnly || Boolean(editorContext.name && editorContext.mode === Constants.routes.baseCreateRoute)}}
                 value={value?.toLowerCase()}
                 id={id}
                 editorContext={editorContext}
-                onChange={onChange}
+                onChange={handleOnChange}
```

Comment:
can't we have the same behaviour for Field onChange and Button onClick .. if one copy past a long phrase to the field, the limitation characters should be triggered too.

Reply:
The field size is limited to the jahia.jcr.maxNameSize so the limitation will be triggered on a copy paste